src/hooks/useGetAllReleases.ts

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { PipelineRunLabel } from '../consts/pipelinerun';
+import { K8sQueryListResourceItems } from '../k8s';
+import { ReleaseModel } from '../models';
+import { ReleaseKind } from '../types';
+
+export interface Release {
+  releasePlan: string;
+  application: string;
+  tenant: string;
+  name: string;
+  completionTime: string;
+  status: string;
+  component: string;
+}
+
+/**
+ * Get all the valid releases across the namespaces that you can see.
+ *
+ * @returns the array of the Release interface.
+ */
+export const useGetAllReleases = (loaded, namespaces) => {
+  const [releases, setData] = React.useState<Release[]>([]);
+
+  React.useEffect(() => {
+    const fetchAll = async (): Promise<Release[]> => {
+      const results: Release[] = [];
+      const namespacePromises = namespaces.map(async (namespace) => {
+        const nsReleases = await K8sQueryListResourceItems<ReleaseKind[]>({
```

Comment:
Releases will be archived using kubearchive going forward. We need to fetch from kubearchive also

Reply:
I am thinking of to fetch the active releases in the current page, fetching the archive ones should be another topic.